### PR TITLE
logind: some minor refactoring around logind's HandleAction logic

### PR DESCRIPTION
**Pull request by poettering**
Just some renaming mostly, to make things more systematic.

/cc @lnussel

### CODE REVIEW

File: src/login/logind-action.c [hunk 1]
```diff
@@ -113,11 +113,12 @@ static const HandleActionData action_table[_HANDLE_ACTION_MAX] = {
         },
 };
 
-const HandleActionData* manager_item_for_handle(HandleAction handle) {
-        assert(handle >= 0);
-        assert(handle < (ssize_t) ELEMENTSOF(action_table));
+const HandleActionData* handle_action_lookup(HandleAction action) {
 
-        return &action_table[handle];
+        if (action < 0 || (size_t) action >= ELEMENTSOF(handle_action_data_table))
```
Comment (bluca) [inline src/login/logind-action.c:118]: This is more than renaming, it's a slight change in behaviour - which is fine, but please mention it in the commit message (commit: `logind: rename manager_item_for_handle() → handle_action_lookup()`)

**Review by bluca — APPROVED** [review]

**Comment by lnussel** [comment]
lgtm

**Comment by poettering** [comment]
made the change @bluca suggested.

**Comment by bluca** [comment]
Mmmh the logind test failed:

```
autopkgtest [16:02:20]: test logind: [-----------------------
 * try-restarting systemd-logind
 * daemon is started
 * loginctl succeeds
 * XDG_SESSION_ID=
1:name=systemd:/user.slice/user-1000.slice/session-1.scope
0::/user.slice/user-1000.slice/session-1.scope
 * process is in session cgroup
Feb 22 15:58:45 autopkgtest systemd-logind[379]: New session 1 of user ubuntu.
Feb 22 16:02:20 autopkgtest systemd[1]: Stopping User Login Management...
Feb 22 16:02:20 autopkgtest systemd[1]: systemd-logind.service: Deactivated successfully.
Feb 22 16:02:20 autopkgtest systemd[1]: Stopped User Login Management.
Feb 22 16:02:20 autopkgtest systemd[1]: Starting User Login Management...
Feb 22 16:02:20 autopkgtest systemd-logind[34859]: Watching system buttons on /dev/input/event0 (Power Button)
Feb 22 16:02:20 autopkgtest systemd-logind[34859]: Watching system buttons on /dev/input/event1 (AT Translated Set 2 k
eyboard)
Feb 22 16:02:20 autopkgtest systemd-logind[34859]: New seat seat0.
Feb 22 16:02:20 autopkgtest systemd[1]: Started User Login Management.
Feb 22 16:02:20 autopkgtest systemd-logind[34859]: New session 1 of user ubuntu.
Fake Lid Switch: /dev/input/event4
P: /devices/virtual/input/input5/event4
N: input/event4
L: 0
E: DEVPATH=/devices/virtual/input/input5/event4
E: SUBSYSTEM=input
E: DEVNAME=/dev/input/event4
E: MAJOR=13
E: MINOR=68
E: USEC_INITIALIZED=250598216
E: ID_INPUT=1
E: ID_INPUT_SWITCH=1
E: TAGS=:power-switch:
E: CURRENT_TAGS=:power-switch:

Feb 22 16:02:20 autopkgtest systemd-logind[34859]: Watching system buttons on /dev/input/event4 (Fake Lid Switch)
Feb 22 16:02:20 autopkgtest systemd-logind[34859]: Lid closed.
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Main process exited, code=dumped, status=11/SEGV
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Failed with result 'core-dump'.
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Scheduled restart job, restart counter is at 1.
Feb 22 16:02:21 autopkgtest systemd[1]: Stopped User Login Management.
Feb 22 16:02:21 autopkgtest systemd[1]: Starting User Login Management...
Feb 22 16:02:21 autopkgtest systemd-logind[34916]: Watching system buttons on /dev/input/event0 (Power Button)
Feb 22 16:02:21 autopkgtest systemd-logind[34916]: Watching system buttons on /dev/input/event1 (AT Translated Set 2 keyboard)
Feb 22 16:02:21 autopkgtest systemd-logind[34916]: Watching system buttons on /dev/input/event4 (Fake Lid Switch)
Feb 22 16:02:21 autopkgtest systemd-logind[34916]: New seat seat0.
Feb 22 16:02:21 autopkgtest systemd[1]: Started User Login Management.
Feb 22 16:02:21 autopkgtest systemd-logind[34916]: New session 1 of user ubuntu.
   waiting for suspend, 30s remaining...
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Main process exited, code=dumped, status=11/SEGV
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Failed with result 'core-dump'.
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Scheduled restart job, restart counter is at 2.
Feb 22 16:02:21 autopkgtest systemd[1]: Stopped User Login Management.
Feb 22 16:02:21 autopkgtest systemd[1]: Starting User Login Management...
Feb 22 16:02:21 autopkgtest systemd-logind[34922]: Watching system buttons on /dev/input/event0 (Power Button)
Feb 22 16:02:21 autopkgtest systemd-logind[34922]: Watching system buttons on /dev/input/event1 (AT Translated Set 2 keyboard)
Feb 22 16:02:21 autopkgtest systemd-logind[34922]: Watching system buttons on /dev/input/event4 (Fake Lid Switch)
Feb 22 16:02:21 autopkgtest systemd-logind[34922]: New seat seat0.
Feb 22 16:02:21 autopkgtest systemd[1]: Started User Login Management.
Feb 22 16:02:21 autopkgtest systemd-logind[34922]: New session 1 of user ubuntu.
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Main process exited, code=dumped, status=11/SEGV
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Failed with result 'core-dump'.
Feb 22 16:02:21 autopkgtest systemd[1]: systemd-logind.service: Scheduled restart job, restart counter is at 3.
Feb 22 16:02:21 autopkgtest systemd[1]: Stopped User Login Management.
Feb 22 16:02:22 autopkgtest systemd[1]: Starting User Login Management...
Feb 22 16:02:22 autopkgtest systemd-logind[34928]: Watching system buttons on /dev/input/event0 (Power Button)
Feb 22 16:02:22 autopkgtest systemd-logind[34928]: Watching system buttons on /dev/input/event1 (AT Translated Set 2 keyboard)
Feb 22 16:02:22 autopkgtest systemd-logind[34928]: Watching system buttons on /dev/input/event4 (Fake Lid Switch)
Feb 22 16:02:22 autopkgtest systemd-logind[34928]: New seat seat0.
Feb 22 16:02:22 autopkgtest systemd[1]: Started User Login Management.
Feb 22 16:02:22 autopkgtest systemd-logind[34928]: New session 1 of user ubuntu.
Feb 22 16:02:22 autopkgtest systemd[1]: systemd-logind.service: Main process exited, code=dumped, status=11/SEGV
Feb 22 16:02:22 autopkgtest systemd[1]: systemd-logind.service: Failed with result 'core-dump'.
Feb 22 16:02:22 autopkgtest systemd[1]: systemd-logind.service: Scheduled restart job, restart counter is at 4.
Feb 22 16:02:22 autopkgtest systemd[1]: Stopped User Login Management.
Feb 22 16:02:22 autopkgtest systemd[1]: Starting User Login Management...
Feb 22 16:02:22 autopkgtest systemd-logind[34935]: Watching system buttons on /dev/input/event0 (Power Button)
Feb 22 16:02:22 autopkgtest systemd-logind[34935]: Watching system buttons on /dev/input/event1 (AT Translated Set 2 keyboard)
Feb 22 16:02:22 autopkgtest systemd-logind[34935]: Watching system buttons on /dev/input/event4 (Fake Lid Switch)
Feb 22 16:02:22 autopkgtest systemd-logind[34935]: New seat seat0.
Feb 22 16:02:22 autopkgtest systemd[1]: Started User Login Management.
Feb 22 16:02:22 autopkgtest systemd-logind[34935]: New session 1 of user ubuntu.
Feb 22 16:02:22 autopkgtest systemd[1]: systemd-logind.service: Main process exited, code=dumped, status=11/SEGV
Feb 22 16:02:22 autopkgtest systemd[1]: systemd-logind.service: Failed with result 'core-dump'.
Feb 22 16:02:22 autopkgtest systemd[1]: systemd-logind.service: Scheduled restart job, restart counter is at 5.
Feb 22 16:02:22 autopkgtest systemd[1]: Stopped User Login Management.
Feb 22 16:02:22 autopkgtest systemd[1]: Starting User Login Management...
Feb 22 16:02:22 autopkgtest systemd-logind[34940]: Watching system buttons on /dev/input/event0 (Power Button)
Feb 22 16:02:22 autopkgtest systemd-logind[34940]: Watching system buttons on /dev/input/event1 (AT Translated Set 2 keyboard)
Feb 22 16:02:22 autopkgtest systemd-logind[34940]: Watching system buttons on /dev/input/event4 (Fake Lid Switch)
Feb 22 16:02:22 autopkgtest systemd-logind[34940]: New seat seat0.
Feb 22 16:02:22 autopkgtest systemd[1]: Started User Login Management.
Feb 22 16:02:22 autopkgtest systemd-logind[34940]: New session 1 of user ubuntu.
Feb 22 16:02:23 autopkgtest systemd[1]: systemd-logind.service: Main process exited, code=dumped, status=11/SEGV
Feb 22 16:02:23 autopkgtest systemd[1]: systemd-logind.service: Failed with result 'core-dump'.
Feb 22 16:02:23 autopkgtest systemd[1]: systemd-logind.service: Scheduled restart job, restart counter is at 6.
Feb 22 16:02:23 autopkgtest systemd[1]: Stopped User Login Management.
Feb 22 16:02:23 autopkgtest systemd[1]: systemd-logind.service: Start request repeated too quickly.
Feb 22 16:02:23 autopkgtest systemd[1]: systemd-logind.service: Failed with result 'core-dump'.
Feb 22 16:02:23 autopkgtest systemd[1]: Failed to start User Login Management.
   waiting for suspend, 25s remaining...
   waiting for suspend, 20s remaining...
   waiting for suspend, 15s remaining...
   waiting for suspend, 10s remaining...
   waiting for suspend, 5s remaining...
   waiting for suspend, 0s remaining...
closing lid did not cause suspend
autopkgtest [16:02:52]: test logind: -----------------------]
autopkgtest [16:02:52]: test logind:  - - - - - - - - - - results - - - - - - - - - -
logind               FAIL non-zero exit status 1
autopkgtest [16:02:52]: test logind:  - - - - - - - - - - stderr - - - - - - - - - -
closing lid did not cause suspend
```

**Comment by bluca** [comment]
Fails in both focal-amd64 and focal-arm64 in the same way

File: src/login/logind-action.h [hunk 2]
```diff
@@ -21,12 +21,14 @@ typedef enum HandleAction {
...
 typedef struct ActionTableItem ActionTableItem;
 
-#define handle_action_valid(x) (x && (x < _HANDLE_ACTION_MAX))
-
 #include "logind-inhibit.h"
 #include "logind.h"
 #include "sleep-config.h"
 
+static inline bool handle_action_valid(HandleAction a) {
+        return a >= 0 && a < _HANDLE_ACTION_MAX;
```
Comment (yuwata) [inline src/login/logind-action.h:29]: This changes the behavior, previously, `a == 0` is not allowed. And I guess that causes the CI failure.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/login/logind-action.h (hunk 2 above)
Comment (lnussel) [reply src/login/logind-action.h:29]: the logic where the function is called is flawed. Was part of the refactoring, no idea how that happened. Sorry for that :-(
Fix:
https://github.com/systemd/systemd/pull/22522/commits/8c65875a9009b5f552893ef1f54ad2261c12f7ad

Comment (lnussel) [reply src/login/logind-action.h:29]: now https://github.com/systemd/systemd/pull/22522/commits/54616cfa5c843b2d53ba2a36f50de98a94fefc37, should be cherry-pickable

**Comment by keszybz** [comment]
Let's merge #22522 first, and then rebase this on top.

**Comment by poettering** [comment]
force pushed a new rebased version. PTAL.

File: src/login/logind-action.c [hunk 3]
```diff
@@ -113,11 +113,12 @@ static const HandleActionData action_table[_HANDLE_ACTION_MAX] = {
...
+                return NULL;
```
Comment (bluca) [inline src/login/logind-action.c:119]: now that this can return null, have you checked that all callers check the result before dereffing it?

Comment (yuwata) [reply src/login/logind-action.c:119]: If it returns NULL, it previously triggers a assertion. Then, that should be ok.

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**